middle_end/flambda2/tests/mlexamples/arrays_compatibility.ml

```diff
@@ -0,0 +1,33 @@
+(* Example from comby: an array access was improperly simplified to Invalid.
+
+   The issue was that the array is seen as an array of immediates by the
+   primitive, but its type from the environment is an array of values. The two
+   kinds are compatible, so the array access is valid. *)
```

Comment:
I don't understand. Surely the access _isn't_ in general valid? All we know from the environment is that it's an array of values, which could be an array of blocks.

Reply:
It depends on the definition of "valid". Here I mean it as "not provably invalid". The only case were we are allowed to return `Invalid` is when then access is provably invalid (no value of the given array type would be allowed as argument of the primitive).

Reply:
Ah, gotcha.